public/pages/workflow_detail/tools/resources/resource_list_with_flyout.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState, useEffect } from 'react';
+import {
+  Direction,
+  EuiCodeBlock,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutHeader,
+  EuiInMemoryTable,
+  EuiTitle,
+  EuiIcon,
+  EuiText,
+} from '@elastic/eui';
+import {
+  Workflow,
+  WorkflowResource,
+  customStringify,
+} from '../../../../../common';
+import { fetchResourceData } from '../../../workflows/workflow_list/resource_list';
```

Comment:
If you'd like to reuse this fn, let's lift it up to a shared utils file somewhere so then it is logically consumed by the downstream components.